src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/service/NetworkWatchListService.java

```diff
@@ -73,27 +84,74 @@ public void removeWatch(Endpoint endpoint) {
         endpoints.remove(endpoint);
     }
 
+    /**
+     * Handle update slow discovery poll flag request.
+     */
+    public void updateSlowPoll(Endpoint endpoint, boolean isEnabled) {
+        log.info("Watch-list service receive update slow poll flag to {} request for {}", isEnabled, endpoint);
+        endpoints.computeIfPresent(endpoint, (e, endpointContext) ->
+                endpointContext.toBuilder().slowPollEnabled(isEnabled).build());
+    }
+
+    @VisibleForTesting
+    void updateUnhurriedPoll(Endpoint endpoint, boolean isEnabled, long currentTime) {
+        log.info("Watch-list service receive update unhurried poll flag to {} request for {}", isEnabled, endpoint);
+        endpoints.computeIfPresent(endpoint, (e, endpointContext) ->
+                endpointContext.toBuilder().unhurriedPollEnabled(isEnabled).build());
+
+        if (!isEnabled) { // when another mechanism used to determine ISL status is disabled
+            for (Set<Endpoint> e : timeouts.values()) {
+                e.remove(endpoint);
+            }
+
+            carrier.discoveryRequest(endpoint, currentTime);
```

Comment:
define some method capable to calculate timeout from endpoint or EndpointContext instance.

Reply:
This method has been added.

Reply:
Why you are doing such timeout's reset during switch from `auxiliary` to `generic` mode, but don't do for `exhausted` to `generic`?